pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -3060,6 +3060,11 @@ let
     (overrideCabal (drv: { src = gogolSrc; }) drv).overrideAttrs (_oldAttrs: {
       sourceRoot = "${gogolSrc.name}/${dir}";
     });
+  isGogolService = name: lib.hasPrefix "gogol-" name && name != "gogol-core";
+  gogolServices = lib.filter isGogolService (lib.attrNames super);
+  gogolServiceOverrides = (lib.foldl' (acc: name:
+    acc // { "${name}" = setGogolSourceRoot "lib/services/${name}" super.${name}; }
+  ) {}) gogolServices;
```

Comment:
Sorry for only reading this in earnest now, but you might want to use https://noogle.dev/f/lib/genAttrs here.

Reply:
Thanks 👍 wasn't aware about getAttrs. Looks a lot cleaner now.